warehouse/api.go

```diff
@@ -87,27 +89,50 @@ type UploadAPIT struct {
 	dbHandle          *sql.DB
 	log               logger.LoggerI
 	connectionManager *controlplane.ConnectionManager
-	isHosted          bool
+	isMultiWorkspace  bool
 }
 
 var UploadAPI UploadAPIT
 
-func InitWarehouseAPI(dbHandle *sql.DB, log logger.LoggerI) {
-	workspaceToken := config.GetWorkspaceToken()
-	isMultiWorkspace := config.GetEnvAsBool("HOSTED_SERVICE", false)
-	if isMultiWorkspace {
-		workspaceToken = config.GetEnv("HOSTED_SERVICE_SECRET", "password")
+func InitWarehouseAPI(dbHandle *sql.DB, log logger.LoggerI) error {
```

Comment:
Can we have a test for this function?

Reply:
Added a test for fetching deployment type. Do we need a function for warehouse functions as well?